buildtest/buildsystem/base.py

```diff
@@ -449,11 +437,12 @@ def run_tests(self, testfile):
         result["BUILD_ID"] = self.build_id
 
         # Change to the test directory
-        os.chdir(self._get_testdir())
-        self.logger.debug(f"Changing to directory {self._get_testdir()}")
+        os.chdir(self.testdir)
+
+        self.logger.debug(f"Changing to test directory {self.testdir}")
 
         # Run the test file using the shell
-        cmd = [self.get_shell(), testfile]
+        cmd = [self.get_shell(), self.metadata["testpath"]]
```

Comment:
If you are going to continually reference self.metadata["testpath"], and removing self._get_testpath, a much more elegant solution is to create a property that essentially returns the same thing, e.g.,

```python
@property
def testpath(self, testpath=None):
        """Based on the testfile path, return the testing directory.
           Returns: full path to testing directory
        """
       return self.metadata["testpath"]
```
You could also combine this sort of thing with being able to parse the original testpath. Anyway, the way you address the issue is up to you, this is a separate suggestion to make reference of the variable much cleaner than self.metadata every time.